core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -312,6 +312,7 @@ static Builder builderFor(FileIO io, Iterable<ManifestFile> deleteManifests) {
   static class Builder {
     private final FileIO io;
     private final Set<ManifestFile> deleteManifests;
+    private long minSequenceNumber = 0L;
```

Comment:
I remember the sequence number 0 is kept for the data files for iceberg v1, so in theory the sequence number from delete files should start from 1.   So setting it to 0 as the default value sounds correct.